### PR TITLE
Feature: Update MFTF to 2.6.0, allow updates to patch versions

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Update **Magento Functional Testing Framework** to the latest version (`2.6.0`). Allow updates of the patch (backward compatible versions).

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

### Questions or comments
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by ihor-sviziev — APPROVED** [review]
Failing integration tests in EE and b2b versions definitely related to updating mftf framework .

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6832](https://jira.corp.magento.com/browse/ENGCOM-6832) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by gwharton** [comment]
Arghh. Sorry. I hit that by accident.

File: composer.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
         "friendsofphp/php-cs-fixer": "~2.14.0",
         "lusitanian/oauth": "~0.8.10",
         "magento/magento-coding-standard": "*",
-        "magento/magento2-functional-testing-framework": "2.5.4",
+        "magento/magento2-functional-testing-framework": "^2.6.0",
```
Comment (fooman) [inline composer.json]: ^ does not match the title of this PR

^ would allow upgrades up til 3.0.0
~2.6.0 would allow all 2.6.* versions

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz,
Could you update dependency to allow only patch releases ?

File: composer.json (hunk 1 above)
Comment (lbajsarowicz) [reply composer.json]: You are right. Thanks for noticing that fact.

**Comment by lbajsarowicz** [comment]
@fooman / @ihor-sviziev Done.

**Comment by ihor-sviziev** [comment]
Failing functional tests looks not related, but to make sure I restarted tests again

**Review by ihor-sviziev — APPROVED** [review]
Failing functional tests not related to this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6832](https://jira.corp.magento.com/browse/ENGCOM-6832) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

**Comment by engcom-Echo** [comment]
Failed functional test not related to the changes in this PR test fails in 2.4-develop with MSI modules.
image

**Comment by slavvka** [comment]
@lbajsarowicz MFTF has been upgraded by an internal team. Closing it.

**Comment by ihor-sviziev** [comment]
@slavvka this PR was not only about update to latest version, but also allow updates to new patch versions! Is there any reason to have exact version installed?

Anyway `composer install` works only with `composer.lock` file, so until you update it - all tests will use version that listed in this file, even when new version will be available, so all tests will not start failing when new version will be released.

**Comment by slavvka** [comment]
@ihor-sviziev The latest update was related to upgrade of MFTF itself - support of securities. I think MFTF changes may be strictly tied to our testing infra so we can’t just take the latest version. But it would be good to address your question to MFTF team. @okolesnyk What do you think?

**Comment by ihor-sviziev** [comment]
@slavvka @okolesnyk I was talking about allow patch releases only. Even minor versions might cause the issues.

**Comment by slavvka** [comment]
@ihor-sviziev It could make sense if we didn't have `composer.lock` file. But since we do any MFTF version bump will require a code change.

**Comment by ihor-sviziev** [comment]
@slavvka make sense. Thank you!

**Comment by ihor-sviziev** [comment]
@okolesnyk thank you so much for adding an ability to install patch versions of mftf https://github.com/magento/magento2/commit/f8da9f14d88e2ee750ad46eda58171f13bc7a064 !

**Pull request closed without merging**